src/api/CMakeLists.txt

```diff
@@ -53,7 +53,7 @@ if (BUILD_SHARED_LIBS)
         LIBRARY DESTINATION lib
         ARCHIVE DESTINATION lib
         RUNTIME DESTINATION bin
-        INCLUDES DESTINATION include
+        INCLUDES DESTINATION include ${INSTALL_HEADERS_DIR}
```

Comment:
From CMake docs (https://cmake.org/cmake/help/latest/command/install.html):

`INCLUDES DESTINATION`
This option specifies a list of directories which will be added to the `INTERFACE_INCLUDE_DIRECTORIES` target property of the `<targets>` when exported by the `install(EXPORT)` command. If a relative path is specified, it is treated as relative to the `$<INSTALL_PREFIX>`.

This means that when another project imports OpenSMT target using `CMake`'s packaging mechanism, these directories will be added as the include directories for that target, and any target that will depend on OpenSMT will inherit these include directories.

`${INSTALL_HEADERS_DIR}` are set to `install/opensmt`,  so that means that a project using OpenSMT as a library with CMake can include `OpenSMT's` headers without the `opensmt` prefix.

Reply:
I am actually now not sure if this would be a good thing. But it should also not be a problem, no?

Reply:
Based on [this](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-incform), I think that our preference of `#include ""` convention over `<>` should be fine. I am not sure though what could be the consequences using such CMake.
For example, I am not sure what will happen if project `A` wants to use `opensmt` as a library when both projects use their own `dir/header.h` and include it this way without a prefix. I think it would be best for now to postpone the problem, merge the PR and maybe to quickly test it within a parent project (e.g. AI expl.) if such a name clash would result in an error. If not, perfect. Otherwise I think we should fix this somehow. Although I am a bit surprised that this is not discussed in the provided link, so maybe it won't be a problem.

Reply:
.. after additional research, the form where we omit prefix `opensmt/` within our project should work fine. It is then the responsibility of the super-project to set the include paths correctly and to follow the convention `#include <opensmt/...`

Reply:
After some discussion, I think we agreed that adding `${INSTALL_HEADERS_DIR}` automatically to include directories of the target is not a good idea :) We want the default to force users to use `opensmt` prefix when including our headers. They can customize the behaviour in their CMake if desirable.

Reply:
It turns out this is not required at the moment.
But it has some undesirable effects, so we will have to rethink this again.